Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,196 @@
+/// </summary>
+public static class HttpContextExtensions
+{
+    /// <summary>
+    /// Translates an <see cref="HttpContext"/> to an <see cref="APIGatewayHttpApiV2ProxyRequest"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="HttpContext"/> to be translated.</param>
+    /// <param name="apiGatewayRouteConfig">The configuration of the API Gateway route, including the HTTP method, path, and other metadata.</param>
+    /// <returns>An <see cref="APIGatewayHttpApiV2ProxyRequest"/> object representing the translated request.</returns>
+    public static APIGatewayHttpApiV2ProxyRequest ToApiGatewayHttpV2Request(
+        this HttpContext context,
+        ApiGatewayRouteConfig apiGatewayRouteConfig)
+    {
+        var request = context.Request;
+        var currentTime = DateTimeOffset.UtcNow;
+        var body = HttpRequestUtility.ReadRequestBody(request);
+        var contentLength = HttpRequestUtility.CalculateContentLength(request, body);
+
+        var pathParameters = RouteTemplateUtility.ExtractPathParameters(apiGatewayRouteConfig.Path, request.Path);
+
+        // Format 2.0 doesn't have multiValueHeaders or multiValueQueryStringParameters fields. Duplicate headers are combined with commas and included in the headers field.
+        var (_, allHeaders) = HttpRequestUtility.ExtractHeaders(request.Headers);
+        var headers = allHeaders.ToDictionary(
+            kvp => kvp.Key,
+            kvp => string.Join(", ", kvp.Value)
+        );
+
+        // Duplicate query strings are combined with commas and included in the queryStringParameters field.
+        var (_, allQueryParams) = HttpRequestUtility.ExtractQueryStringParameters(request.Query);
+        var queryStringParameters = allQueryParams.ToDictionary(
+            kvp => kvp.Key,
+            kvp => string.Join(",", kvp.Value)
+        );
+
+        string userAgent = request.Headers.UserAgent.ToString();
+
+        if (!headers.ContainsKey("content-length"))
+        {
+            headers["content-length"] = contentLength.ToString();
+        }
+
+        if (!headers.ContainsKey("content-type"))
+        {
+            headers["content-type"] = "text/plain; charset=utf-8";
+        }
+
+        var httpApiV2ProxyRequest = new APIGatewayHttpApiV2ProxyRequest
+        {
+            RouteKey = $"{request.Method} {apiGatewayRouteConfig.Path}",
+            RawPath = request.Path.Value, // this should be decoded value
+            Body = body,
+            IsBase64Encoded = false,
+            RequestContext = new ProxyRequestContext
+            {
+                Http = new HttpDescription
+                {
+                    Method = request.Method,
+                    Path = request.Path.Value, // this should be decoded value
+                    Protocol = !string.IsNullOrEmpty(request.Protocol) ? request.Protocol : "HTTP/1.1", // defaults to http 1.1 if not provided
```

Comment:
should we make the "HTTP/1.1" assumption? This is more likely to get updated and we wouldn't know to update it

Reply:
ive put this here so it matches api gateway